omnibus/files/private-chef-cookbooks/private-chef/templates/default/haproxy.cfg.erb

```diff
@@ -0,0 +1,26 @@
+global
+        stats socket /var/opt/opscode/haproxy/haproxy.sock mode 600
+
+frontend postgresql
+         bind <%= @listen %>:<%= @local_postgresql_port %>
+         mode tcp
+         default_backend chef_backend_postgresql
+
+frontend elasticsearch
+         bind <%= @listen %>:<%= @local_elasticsearch_port %>
+         mode tcp
+         default_backend chef_backend_elasticsearch
+
+backend chef_backend_postgresql
+        mode tcp
+        option httpchk GET /leader HTTP/1.1\r\nHost:localhost:<%= @leaderl_healthcheck_port %>\r\n\r\n
+        <% @chef_backend_members.each do |name, ip| -%>
+        server <%= name %> <%= ip %>:<%= @remote_postgresql_port %> check port <%= @leaderl_healthcheck_port %> rise 1 fall 1
```

Comment:
We are using an HTTP check here because the /leader endpoint knows about the leader/follower state.  (postgresql could be erroneously up as a leader while leaderl is in the process of taking it down.  The /leader endpoint will return the correct thing in that case while a postgresql-based check may not).